Tests/MusicXMLTests/ScoreTests/ScoreTestSuite.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  ScoreTestSuite.swift
+//  MusicXMLTests
+//
+//  Created by Ben Lu on 10/14/19.
+//
+
+import XCTest
+import MusicXML
+
+class ScoreTests: XCTestCase {
+
+    func testAll() throws {
+        let paths = try FileManager.default.contentsOfDirectory(atPath: testSuiteURL.path)
+        for path in paths {
+            let testCaseURL = testSuiteURL.appendingPathComponent(path)
+            guard directoryExists(at: testSuiteURL.path) else {
+                continue
+            }
+            let manifestURL = testSuiteURL.appendingPathComponent(path).appendingPathComponent("manifest.plist")
+            guard let manifestData = FileManager.default.contents(atPath: manifestURL.path) else {
+                continue
+            }
+            let manifest = try PropertyListDecoder().decode(ScoreTestCaseManifest.self, from: manifestData)
+            try runTest(forManifest: manifest, testCaseURL: testCaseURL)
+        }
+    }
+
+    private func runTest(forManifest manifest: ScoreTestCaseManifest, testCaseURL: URL) throws {
+        let musicXMLURL = testCaseURL.appendingPathComponent(manifest.xmlFileName)
+
+        print("🎼 Test score: \(manifest.displayName)")
+        print("  - Path: \(musicXMLURL.path)")
+        if let description = manifest.description {
+            print("  - Description: \(description)")
+        }
+        if let source = manifest.source {
+            print("  - Source: \(source)")
+        }
+
+        var parsed: MusicXML?
+        do {
+            parsed = try MusicXML(url: musicXMLURL)
+            print("  - ✅ Parsing succeeded")
+        } catch {
+            print("  - ❌ Parsing failed")
+        }
+
+        guard let className = manifest.expectationClassName, let clazz = Bundle(for: Self.self).classNamed("MusicXMLTests.\(className)"), let expectationClazz = clazz as? ScoreTestExpectation.Type else {
+            print("  - ❓ Expectation class not found. Make sure to specify SwiftClassName in the manifest and make the class conform to ScoreTestExpectation.")
+            return
+        }
+
+        if let parsed = parsed {
+            let expected = expectationClazz.expectedMusicXML
+
+            if parsed == expected {
+                print("  - ✅ Compared equal to expectation")
+            } else {
+                print("  - ❌ Not equal to expectation")
+                #warning("TODO: Maybe generate a diff")
```

Comment:
I _really_ like the idea of a diff. Currently, sniffing out differences between `MusicXML` values is an exceptional PIA.

Could [ordered collection diffing](https://github.com/apple/swift-evolution/blob/master/proposals/0240-ordered-collection-diffing.md) be of use here?

Reply:
Looking into the ordered diffing, it looks like the answer is no. Perhaps `Mirror`? There must be a way 🕵 .

Reply:
I can think we may be able to use Mirror to get a deep recursive comparison between objects. I found no such tools in the market at the moment.

Reply:
Ooh, we could use [swift-snapshot-testing](https://github.com/pointfreeco/swift-snapshot-testing/blob/master/Documentation/Available-Snapshot-Strategies.md#dump).